codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -153,11 +174,15 @@ private static ByteBuf encodeConnectMessage(
         buf.writeByte(variableHeader.version());
         buf.writeByte(getConnVariableHeaderFlag(variableHeader));
         buf.writeShort(variableHeader.keepAliveTimeSeconds());
+        buf.writeBytes(propertiesBuf);
+        propertiesBuf.release();
 
         // Payload
         buf.writeShort(clientIdentifierBytes.length);
         buf.writeBytes(clientIdentifierBytes, 0, clientIdentifierBytes.length);
         if (variableHeader.isWillFlag()) {
+            buf.writeBytes(willPropertiesBuf);
+            willPropertiesBuf.release();
```

Comment:
I wonder if we should better do the release in a finally block. Just in case if something goes wrong... otherwise we will leak memory

Reply:
Also I think we should always call `release()` ... so please do it outside of the if block. I know at the moment it not really matters as you use `EMPTY_BUFFER` but this is more of an implementation detail.